### PR TITLE
improvement: S3C-2290: Upgrade to node 10

### PR DESCRIPTION
**Pull request by Katherinelaue1**

### CODE REVIEW

**Comment by Katherinelaue1** [comment]
@bert-e reset

File: lib/IAMClient.js [hunk 1]
```diff
@@ -264,16 +268,19 @@ class VaultClient {
...
+            assert.notStrictEqual(conv || marker === '', true,
                 'Marker must be a number');
             assert(Number(marker) >= 0, 'Marker must be >= 0');
             data.Marker = marker;
         }
         if (typeof options.maxItems !== 'undefined') {
             assert.strictEqual(typeof options.maxItems, 'number',
                 'maxItems need to be a number');
-            assert.notStrictEqual(isNaN(options.maxItems), true,
+            // eslint-disable-next-line no-restricted-globals
```
Comment (anurag4DSB) [inline lib/IAMClient.js]: This line is not needed can be removed?
~~// eslint-disable-next-line no-restricted-globals~~

File: package.json [hunk 2]
```diff
@@ -14,23 +14,24 @@
   },
   "homepage": "https://github.com/scality/vaultclient#readme",
   "dependencies": {
-    "arsenal": "scality/Arsenal#67365083",
-    "commander": "2.9.0",
-    "xml2js": "0.4.17",
-    "werelogs": "scality/werelogs#0ff7ec82"
+    "arsenal": "scality/Arsenal#4fd77d73",
```
Comment (anurag4DSB) [inline package.json]: We have referred to the wrong commit. We will need to change to this

File: tests/unit/getAccountPropertiesTests.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
+/* eslint-disable operator-linebreak */
 'use strict'; // eslint-disable-line
 
 const assert = require('assert');
 const http = require('http');
-const IAMClient = require('../../lib/IAMClient.js');
 const querystring = require('querystring');
+const IAMClient = require('../../lib/IAMClient.js');
+
```
Comment (anurag4DSB) [inline tests/unit/getAccountPropertiesTests.js:8]: We can get rid of the extra new line here

File: tests/unit/v2authTests.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
 'use strict'; // eslint-disable-line
 
-const errors = require('arsenal').errors;
+const { errors } = require('arsenal');
 const assert = require('assert');
 const http = require('http');
-const IAMClient = require('../../lib/IAMClient.js');
 const querystring = require('querystring');
+const IAMClient = require('../../lib/IAMClient.js');
+
```
Comment (anurag4DSB) [inline tests/unit/v2authTests.js:8]: Can get rid of this extra line as well

File: tools/blaster.js [hunk 5]
```diff
@@ -140,10 +141,10 @@ function createAccount(index, callback) {
                 usersEmails: [],
             };
             return callback(null, index + 1);
-        } else if (error && err && err.EntityAlreadyExists) {
+        } if (error && err && err.EntityAlreadyExists) {
```
Comment (anurag4DSB) [inline tools/blaster.js]: This `if ()` block needs to be on the new line.

File: tools/blaster.js [hunk 6]
```diff
@@ -203,18 +204,20 @@ function createUser(index, callback) {
...
-    const params = {email, password: getPassword()};
+    const params = { email, password: getPassword() };
     client.createUser(accountName, userName, params, (err, data) => {
         if (!error && !err) {
             const acc = accounts[accountName];
             acc.usersNames.push(userName);
             acc.usersEmails.push(email);
-            acc.users[userName] = {email, data, accessKeys: [], secretKeys: {}};
+            acc.users[userName] = {
+                email, data, accessKeys: [], secretKeys: {},
```
Comment (anurag4DSB) [inline tools/blaster.js]: Make is more readable:
```suggestion
                email,
                data,
                accessKeys: [],
                secretKeys: {},
```

File: tools/blaster.js [hunk 7]
```diff
@@ -203,18 +204,20 @@ function createUser(index, callback) {
...
+            };
             return callback(null, index + 1);
-        } else if (error && err) {
+        } if (error && err) {
```
Comment (anurag4DSB) [inline tools/blaster.js]:
```suggestion
        } 
        if (error && err) {
```

Comment (anurag4DSB) [reply tools/blaster.js]: Next `if()` block needs to be on a new line

File: tools/blaster.js [hunk 8]
```diff
@@ -243,18 +246,18 @@ function deleteUser(index, callback) {
             const acc = accounts[accountName];
             acc.usersNames.splice(acc.usersNames.indexOf(userName), 1);
             acc.usersEmails.splice(
-                acc.usersEmails.indexOf(acc.users[userName].email), 1);
+                acc.usersEmails.indexOf(acc.users[userName].email), 1,
+            );
             acc.users[userName] = undefined;
             return callback(null, index + 1);
-        } else if (error && err) {
+        } if (error && err) {
```
Comment (anurag4DSB) [inline tools/blaster.js]: Same here, `if` block needs to be on a new line

File: tools/blaster.js [hunk 9]
```diff
@@ -279,13 +282,13 @@ function createAccessKey(index, callback) {
             const accessKey = data.message.body.id;
             const secretKey = data.message.body.value;
             accounts[accountName].users[userName]
-                    .secretKeys[accessKey] = secretKey;
+                .secretKeys[accessKey] = secretKey;
             accounts[accountName].users[userName].accessKeys.push(accessKey);
             return callback(null, index + 1);
-        } else if (err && error) {
+        } if (err && error) {
```
Comment (anurag4DSB) [inline tools/blaster.js]: same here

File: tools/blaster.js [hunk 10]
```diff
@@ -311,15 +314,15 @@ function deleteAccessKey(index, callback) {
...
-            const accessKeys = accounts[accountName].users[userName].accessKeys;
+            const { accessKeys } = accounts[accountName].users[userName];
             accessKeys.splice(accessKeys.indexOf(accessKey), 1);
-            accounts[accountName].
-                    users[userName].secretKeys[accessKey] = undefined;
+            accounts[accountName]
+                .users[userName].secretKeys[accessKey] = undefined;
             return callback(null, index + 1);
-        } else if (err && error) {
+        } if (err && error) {
```
Comment (anurag4DSB) [inline tools/blaster.js]: same here, new line for if block

File: tools/blaster.js [hunk 11]
```diff
@@ -355,10 +358,10 @@ function verifySignatureV2(index, callback) {
     client.verifySignatureV2('signature', signature, accessKey, params, err => {
         if (!error && !err) {
             return callback(null, index + 1);
-        } else if (error && err) {
+        } if (error && err) {
```
Comment (anurag4DSB) [inline tools/blaster.js]: same here, new line for if block

File: package.json (hunk 2 above)
Comment (anurag4DSB) [reply package.json]: The correct tags for Arsenal are:
development/7.4 - 9f2e74e
development/7.5 - b03f5b8
development/8.0 - dd6fde6
development/8.1 -  c57cde8

File: package.json [hunk 12]
```diff
@@ -14,23 +14,24 @@
...
+    "commander": "2.20.0",
+    "xml2js": "0.4.19",
+    "werelogs": "scality/werelogs#4e0d97c"
```
Comment (anurag4DSB) [inline package.json:20]: The correct tags for Werelog are:
development/7.4 - 4e0d97c
development/7.5 - 4e0d97c
development/8.0 - 351a2a3  -> needs to be fixed in the integration branch
development/8.1 - 351a2a3   -> needs to be fixed in the integration branch

File: package.json (hunk 2 above)
Comment (anurag4DSB) [reply package.json]: All 4 have wrong commits.
Needs to be fixed.

**Review by anurag4DSB — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Katherinelaue1) [reply package.json]: Updated development/7.4 to 9f2e74e. Will follow up with the integration branches before merging.

File: package.json (hunk 12 above)
Comment (Katherinelaue1) [reply package.json:20]: Updated development/7.4 to 4e0d97c . Will follow up with the integration branches before merging.

**Review by anurag4DSB — APPROVED** [review]
LGTM.
Just make sure the integration branches have the correct tags and updates changes.

**Review by JianqinWang — APPROVED** [review]

**Comment by JianqinWang** [comment]
LGTM, it would be helpful to squash commits before merge

File: lib/IAMClient.js [hunk 13]
```diff
@@ -1,12 +1,13 @@
+/* eslint-disable no-underscore-dangle */
 'use strict'; // eslint-disable-line
 
-const errors = require('arsenal').errors;
-const auth = require('arsenal').auth;
+const { errors } = require('arsenal');
```
Comment (tmacro) [inline lib/IAMClient.js]: perhaps 
```javascript
const { auth, errors } = require('arsenal');
```

**Review by tmacro — APPROVED** [review]

File: lib/IAMClient.js (hunk 13 above)
Comment (Katherinelaue1) [reply lib/IAMClient.js]: Changed it, thanks for idea!

**Comment by Katherinelaue1** [comment]
@bert-e reset

**Comment by Katherinelaue1** [comment]
@bert-e force_reset

**Review by anurag4DSB — APPROVED** [review]

**Comment by Katherinelaue1** [comment]
@bert-e approve

**Review by bennettbuchanan — APPROVED** [review]

**Pull request merged**